python/GafferImageUI/ImageViewUI.py

```diff
@@ -519,6 +475,57 @@ def __setValue( self, value, *unused ) :
 
 		self.getPlug().setValue( value )
 
+##########################################################################
+# _LutGPUPlugValueWidget
+##########################################################################
+
+class _LutGPUPlugValueWidget( GafferUI.PlugValueWidget ) :
+
+	def __init__( self, plug, **kw ) :
+
+		self.__button = GafferUI.MenuButton(
+			image = "lutGPU.png",
+			hasFrame = False,
+			menu = GafferUI.Menu(
+				Gaffer.WeakMethod( self.__menuDefinition ),
+				title = "DisplayTransform Mode",
+			)
+		)
+
+		GafferUI.PlugValueWidget.__init__( self, self.__button, plug, **kw )
+
+		self._updateFromPlug()
+
+	def _updateFromPlug( self ) :
+
+		with Gaffer.Context() :
+
+			self.__button.setImage( "lutGPU.png" if self.getPlug().getValue() else "lutCPU.png" )
```

Comment:
When switching to GPU, I'm not seeing the image update immediately. It seems a redraw isn't requested until after I subsequently move the mouse. This is on Linux. Do you see the same?

Reply:
This my main remaining concern at this point - I haven't been able to reproduce this no matter how much I make sure the cursor isn't moving, and I'm not sure what would be happening.

Reply:
I've spent a bit longer digging into this, after rebasing on master to be sure there wasn't any interaction with recent button highlighting fixes. I've now convinced myself that it has nothing whatsoever to do with mouse movement, and it's just that the first redraw is a little slow when switching to GPU mode. VTune implies that the delay is just OCIO generating the LUT, so I don't think we should worry about this. Sorry for the bother.

What is odd is that there is a redraw reported after we update the icon and before the redraw that pays for the LUT. So I don't know why Qt hasn't updated the icon visibly at that point, but that's a rabbit hole we definitely should not go down.